src/components/HorizontalStepper/HorizontalStepper.tsx

```diff
@@ -0,0 +1,82 @@
+import clsx from 'clsx';
+import React from 'react';
+import styles from './HorizontalStepper.module.css';
+import HorizontalStep from '../HorizontalStep';
+
+export interface Props {
+  /**
+   * Zero-based index that identifies which index is the active step.
+   */
+  activeIndex: number;
+  /**
+   * CSS class names that can be appended to the component.
+   */
+  className?: string;
+  /**
+   * Ordered list of step texts.
+   */
+  steps: string[];
+}
+
+/**
+ * ```ts
+ * import {HorizontalStepper} from "@chanzuckerberg/eds";
+ * ```
+ *
+ * A stepper component used to display which steps have been completed, the current active step, and possible remaining steps.
+ *
+ * Example usage:
+ *
+ * ```tsx
+ * <HorizontalStepper activeIndex={0} steps={['Step 1', 'Step 2', 'Step 3']} />
+ * ```
+ */
+export const HorizontalStepper = ({ activeIndex, className, steps }: Props) => {
+  if (
+    process.env.NODE_ENV !== 'production' &&
+    !(activeIndex >= 0 && activeIndex <= steps.length)
```

Comment:
Could we use an or operator to avoid the negation? The negatives makes it a littler harder to reason about the logic.
```suggestion
    (activeIndex < 0 || activeIndex > steps.length)
```

Reply:
Also, maybe add a test?

Reply:
Will add test for this and the other `console.warn` in the subcomponent.
Logic wise, I did the negative to account for `NaN` which pass the type check for number, but if you think that's low value, can change to your suggestion
![image](https://user-images.githubusercontent.com/86632227/165600579-93a04dbb-d999-4264-976d-4ae80863378d.png)

Reply:
Sorry for the confusion – I figured it a little bit after leaving my comment. Follow-up though – does the type check not catch it when `activeNumber` is `NaN` since it's now a required `number`?

Reply:
Type check doesn't catch it since `typeof NaN === 'number'`
TS playground showing no error on NaN:
https://www.typescriptlang.org/play?#code/ATBQBcE8AcFNgHIFcC2AjWAnA8mgVrAMbjAC8wA3qCCAIYBcwAdqhpgNygC+oohA9kwDOJZOiy48jMW0lES5KjWANgARgA03XgOGjaCOcWmsJ+eWUrUaqhAa08+gkYiQAbN0fAnxOc8UslG0YWD20gA

Reply:
Cool, thanks for checking. Could you put that comment in the code for future devs?

Reply:
Yes will do